src/Chapter4.hs

```diff
@@ -628,8 +649,9 @@ Can you implement a monad version of AND, polymorphic over any monad?
 
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
-andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+-- Test is failing on "andM (Just False) Nothing = Nothing", expecting "Just False" ?
+andM :: (Monad m) => m Bool -> m Bool -> m Bool 
+andM bool1 bool2 = bool1 >>= \b1 -> (bool2 >>= (\b2 -> pure (b1 && b2)))
```

Comment:
The ordinary `&&` if lazy on the second argument. Meaning that if it meets `False` as the first argument, there is no need to calculate another arg. In `andM` we would like to keep the same behaviour, so the

```
Just False `andM` _
```

should be `Just False` :slightly_smiling_face: 

Think of how you use `if-then-else` instead of `&&` to implement a lazier version

Reply:
Aaah I see. I was assuming it should have to return the `Nothing`-like value. Thanks!